staking/types/validator.go

```diff
@@ -471,31 +471,55 @@ func VerifyBLSKey(pubKey *shard.BlsPublicKey, pubKeySig *shard.BLSSignature) err
 	return nil
 }
 
-func containsHarmonyBlsKeys(blsKeys []shard.BlsPublicKey,
-	hmyAccounts []genesis.DeployAccount) error {
+func containsHarmonyBLSKeys(
+	blsKeys []shard.BlsPublicKey,
+	hmyAccounts []genesis.DeployAccount,
+	epoch *big.Int,
+) error {
 	for i := range blsKeys {
-		if err := matchesHarmonyBlsKey(&blsKeys[i], hmyAccounts); err != nil {
+		if err := matchesHarmonyBLSKey(
```

Comment:
how about also cache across this outer loop?

Reply:
wrong to cache on those because they came from the validator off the street, otherwise cache will explode. Our cache will be always finite and known size